internal/backend/temporalclient/client.go

```diff
@@ -119,6 +122,36 @@ func (c *impl) startDevServer(ctx context.Context, cfg *Config, opts client.Opti
 
 func (c *impl) Temporal() client.Client { return c.client }
 
+func (c *impl) TemporalAddr() (frontend, ui string) {
+	if c.srv == nil {
+		if frontend = c.cfg.HostPort; frontend == "" || frontend == "localhost" {
+			// known temporal defaults.
+			frontend = "localhost:7233"
+			ui = fmt.Sprintf("http://%s:%d", "localhost", 8233)
+		}
+		return
+	}
+
+	frontend = c.srv.FrontendHostPort()
+
+	host, port, err := net.SplitHostPort(frontend)
+	if err != nil {
+		return
+	}
+
+	nport, err := strconv.Atoi(port)
+	if err != nil {
+		return
+	}
+
+	// temporal's default is frontend+1000 for dev server.
+	nport += 1000
```

Comment:
Problem: this is the default, but it's not necessary. Temporal can be started with this flag:

```
--ui-port value
Port for the Web UI (default: --port + 1000, e.g. 8233)
```

Is there a way to get the actual value? If not, I'm not sure guessing is a good idea.

Reply:
Unfortunately, there is no way to get it, so there is no way around it.

Reply:
That's my point: we're only guessing, and we could be wrong, so why give this info to the user if it can mislead them?

Same thing about the scheme: it **may** be HTTP, but it can also be HTTPS if it's remote.

Other better options, in decreasing order of how much I like them:

1. Don't print a UI URL at all
2. Print it, but also say it's a guess
3. Check the guess with an HTTPS query (with HTTP as fallback) to `/healthz` (expect to receive `{"status":"OK"}`), and print it if we get a response, and "Unknown" otherwise

Reply:
it can't be https since it's coming from the dev server - where it will always run locally over http.
since it's always coming from the devserver, it will always be right.

i could do the probe against its address, but is it worth it? i mean it's an amusing idea, but still...

Reply:
The banner is not limited to dev mode, and even in dev mode creating a Temporal dev server is optional not a requirement.